src/enochecker/enochecker.py

```diff
@@ -303,28 +313,54 @@ def run(self, method: Optional[CheckerMethod] = None) -> CheckerResult:
 
         try:
             ret = self._run_method(method)
+            if method == CheckerMethod.EXPLOIT and (type(ret) == str or ret is None):
+                if type(ret) == str:
+                    self.info(
+                        "Checker [{}] executed successfully and returned found flag: {}".format(
+                            method, ret
+                        )
+                    )
+                    return CheckerResult(CheckerTaskResult.OK, flag=cast(str, ret))
+                else:
+                    self.info(
+                        "Checker [{}] did not return a string, assuming Mumble".format(
+                            method
+                        )
+                    )
+                    return CheckerResult(CheckerTaskResult.MUMBLE)
+
             if ret is not None:
-                warnings.warn(
-                    "Returning a result is not recommended and will be removed in the future. Raise EnoExceptions with additional text instead.",
-                    DeprecationWarning,
-                )
-                try:
-                    CheckerTaskResult(ret)
-                except:
-                    self.error(
-                        "Illegal return value from {}: {}".format(self.method, ret)
+                if type(ret) == str and method == CheckerMethod.PUTFLAG:
+                    self.info(
+                        "Checker [{}] executed successfully and returned attack info: {}".format(
+                            method, ret
+                        )
                     )
                     return CheckerResult(
-                        CheckerTaskResult.CHECKER_TASK_RESULT_INTERNAL_ERROR
+                        CheckerTaskResult.OK, attack_info=cast(str, ret)
```

Comment:
More a spec question, but wouldn't it make sense to have `attack_info` be a `json`?

Reply:
might make sense to have this in the specification, but as far as I can recall most CTFs which had attack info simply have a single string

Reply:
discussion moved to https://github.com/enowars/specification/issues/11